### PR TITLE
adds auth routes, user models, server file, dependencies, and test file

### PR DESCRIPTION
**Pull request by cburk2019**

### CODE REVIEW

**Review by JacobKnaack — CHANGES_REQUESTED** [review]
Please pull and update with requested changes!

File: __tests__/authrouter.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    done();
+  });
+
+  it('can get if user has read permissions =', async (done) => {
+    const body = {};
+
+    const token = jwt.sign(users['user'], process.env.SECRET);
+
+    const res = await server.get('/api').set('Authorization', 'Bearer ' + token);
```
[the remark this answers is not in the document]
Comment (cburk2019) [reply __tests__/authrouter.test.js]: Looks like we were able to get this up and going, thank you for your guidance!

File: __tests__/authrouter.test.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    const res = await server.post('/api')
+      .set('Authorization', 'Bearer ' + token).send(body)
+
+    expect(res.statusCode).toBe(201);
+    expect(res.body).toBe(body);
+
+    const token2 = jwt.sign(users['user'], process.env.SECRET);
+    await server.post('/signup').send(users['user']);
+
+    const res2 = await server.post('/api/v2/food')
```
[the remark this answers is not in the document]
Comment (cburk2019) [reply __tests__/authrouter.test.js]: Thank you for guiding me to the promised land!

**Review by JacobKnaack — APPROVED** [review]
Looks good!

**Pull request merged**